app/api/microservice.py

```diff
@@ -0,0 +1,171 @@
+from fastapi import APIRouter ,Depends,HTTPException ,status
+from app.schemas import MicroserviceInDBBase , MicroserviceCreate , MicroserviceTeamScorecardBase ,MicroserviceCreateApi, MicroserviceScoreCardCreate , MicroserviceUpdate, MicroserviceScoreCardUpdate
+from app.crud import CRUDMicroservice, CRUDMicroserviceTeamScorecard ,CRUDTeam, CRUDScoreCard, CRUDMicroserviceScoreCard
+from typing import List
+from app import dependencies 
+from pydantic import BaseModel, Field
+import re
+
+class Value(BaseModel):
+    value: int | bool | float | str = Field(..., title="Value", description="Value of the service")
+
+router = APIRouter()
+
+def format_code(name):
+    code = name.strip()
+    code = code.replace(" ", "-")
+    code = re.sub(r'-+', '-', code)
```

Comment:
praise: good job to strip the string before replacing spaces.
suggestion: What do you think about this `code = re.sub(r'\s+', '-', name.strip())` ?